### PR TITLE
Transformations v1 backward compatibility

### PR DESCRIPTION
**Pull request by sivashanmukh**
A temporary PR to track differences.

### CODE REVIEW

File: util/customTransformer-v1.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      reject(error.message);
+    }
+  });
+  const timeoutPromise = new Promise(resolve => {
+    const wait = setTimeout(() => {
+      clearTimeout(wait);
+      resolve("Timedout");
+    }, 4000);
+  });
+  const result = await Promise.race([executionPromise, timeoutPromise]);
```
Comment (psrikanth88) [inline util/customTransformer-v1.js:50]: should we catch the rejections here and throw error?

File: util/customTransformer-v1.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    stats.gauge("isolate_vm_pool_size", isolatevmPool.size);
+    stats.gauge("isolate_vm_pool_available", isolatevmPool.available);
+    stats.counter("events_into_vm", events.length);
+    const isolateStartWallTime = calculateMsFromIvmTime(
+      isolatevm.isolateStartWallTime
+    );
+    const isolateStartCPUTime = calculateMsFromIvmTime(
+      isolatevm.isolateStartCPUTime
+    );
+    const transformedEvents = await transform(isolatevm, events);
```
Comment (psrikanth88) [inline util/customTransformer-v1.js:78]: catch errors here?

**Review by psrikanth88 — APPROVED** [review]

File: util/customTransformer-v1.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
+const ivm = require("isolated-vm");
+const fetch = require("node-fetch");
+const fs = require("fs");
+const lodashCore = require("lodash/core");
+const _ = require("lodash");
+const stats = require("./stats");
+
+// TODO: Check why these dont work
+const unsupportedFuncNames = [
```
Comment (lokey) [inline util/customTransformer-v1.js:9]: Remove this if not used

File: util/customTransformer-v1.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+      await isolatevm.bootstrapScriptResult.apply(undefined, [
+        isolatevm.fnRef,
+        new ivm.Reference(resolve),
+        sharedTransformationPayload
+      ]);
+    } catch (error) {
+      reject(error.message);
+    }
+  });
+  const timeoutPromise = new Promise(resolve => {
```
Comment (lokey) [inline util/customTransformer-v1.js:41]: Make sure we tested this case using an infinite loop transformation function. It should exit in 4 seconds with timeout error

File: util/customTransformer-v1.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+  events,
+  userTransformation,
+  libraryVersionIds
+) {
+  if (userTransformation.versionId) {
+    const isolatevmPool = await getPool(userTransformation, libraryVersionIds);
+    const isolatevm = await isolatevmPool.acquire();
+    stats.gauge("isolate_vm_pool_size", isolatevmPool.size);
+    stats.gauge("isolate_vm_pool_available", isolatevmPool.available);
+    stats.counter("events_into_vm", events.length);
```
Comment (lokey) [inline util/customTransformer-v1.js:68]: Will be useful to have transformationVersionId tag for these stats.

File: util/ivmPool.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    transformationPoolCache[versionId] = genericPool.createPool(factory, opts);
+    transformationLibraryCache[versionId] = sortedLibrariesIdString;
+    // Added to stop retrying and infinite loop on error
+    // TODO: Figure out if we should we do this
+    transformationPoolCache[versionId].on("factoryCreateError", error => {
+      // eslint-disable-next-line no-underscore-dangle
+      transformationPoolCache[versionId]._waitingClientsQueue
+        .dequeue()
+        .reject(error);
+    });
```
Comment (lokey) [inline util/ivmPool.js:38]: What does the `/customTransform` endpoint return if factory creation fails and how are we handling it in the server?

File: util/ivmFactory.js [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+    console.log(`import from ${spec} failed. Module not found.`);
+    throw new Error(`import from ${spec} failed. Module not found.`);
+  });
+
+  // const base64Script = await isolate.compileScript(base64);
+  // const customScriptRunResult = await customScript.run(context);
+  // console.log(
+  //   "runUserTransform -> customScriptRunResult",
+  //   customScriptRunResult
+  // );
```
Comment (lokey) [inline util/ivmFactory.js:191]: Remove this

**Pull request merged**